cli/azd/pkg/cloud/cloud.go

```diff
@@ -0,0 +1,99 @@
+package cloud
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+)
+
+const (
+	ConfigPath = "cloud"
+
+	AzurePublicName       = "AzureCloud"
+	AzureChinaCloudName   = "AzureChinaCloud"
+	AzureUSGovernmentName = "AzureUSGovernment"
+)
+
+type Cloud struct {
+	Configuration cloud.Configuration
+
+	// The base URL for the cloud's portal (e.g. https://portal.azure.com for
+	// Azure public cloud).
+	PortalUrlBase string
+
+	// The suffix for the cloud's storage endpoints (e.g. core.windows.net for
+	// Azure public cloud). These are well known values and can be found at:
+	// https://<management-endpoint>/metadata/endpoints?api-version=2023-12-01
+	StorageEndpointSuffix string
+}
+
+type Config struct {
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+}
+
+func NewCloud(config *Config) *Cloud {
+	if cloud, err := getNamedCloud(config.Name); err != nil {
+		// TODO: Find a friendly way to surface errors here
+		panic(err)
```

Comment:
This needs to be dome more correctly. `platform.go` has useful examples here. I'll have a look at those. Do not merge until this is fixed.